traversal/producer/GraphProducer.java

```diff
@@ -113,10 +113,11 @@ private synchronized void transition(Queue<VertexMap> queue, ResourceIterator<Ve
     private synchronized void compensate(Queue<VertexMap> queue, int unfulfilled) {
         ResourceIterator<VertexMap> newIter =
                 new GraphIterator(graphMgr, start.next(), procedure, params, filter).distinct(produced);
-        CompletableFuture<Void> asyncJob = unfulfilled > 0
-                ? CompletableFuture.runAsync(() -> job(queue, newIter, unfulfilled), forkJoinPool())
-                : CompletableFuture.completedFuture(null);
-        runningJobs.put(newIter, asyncJob);
+        runningJobs.put(newIter, CompletableFuture.completedFuture(null));
```

Comment:
Currently there's check in the async job on if `runningJobs` contains the iterator. It's possible that this job is started before we put the future into the job. Therefore we always add `CompletableFuture` to the `runningJobs` before we start the job.